src/utils.ts

```diff
@@ -76,7 +76,7 @@ function transformColor(paintColor: mbColorDefinition, func: Function): mbColorD
       const newColor = transformColor(stop[1], func);
       return [stop[0], newColor];
     });
-    const newPaintColor = { ...paintColor, ...stops };
+    const newPaintColor = Object.assign({}, paintColor, { stops });
```

Comment:
Had to revert to `Object.assign` because the other way Typescript compiler was giving errors with the type of `newPaintColor` not being conformat to maplibre spec.